### PR TITLE
Create DA for Access Groups

### PR DESCRIPTION
**Pull request by MatthewLemmond**
### Description
Create a deployable architecture for the access group creation, utilizes the previously created module to define the accesses provided to the groups and allows for customization around things like names of the groups and adding users/service IDs to the groups

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

Add deployable architecture to manage access groups

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: cra-config.yaml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # More info about this file at https://github.com/terraform-ibm-modules/common-pipeline-assets/blob/main/.github/workflows/terraform-test-pipeline.md#cra-config-yaml
 version: "v1"
 CRA_TARGETS:
-  - CRA_TARGET: "examples/access-management" # Target directory for CRA scan. If not provided, the CRA Scan will not be run.
+  - CRA_TARGET: "solutions/access-management" # Target directory for CRA scan. If not provided, the CRA Scan will not be run.
     CRA_IGNORE_RULES_FILE: "cra-tf-validate-ignore-rules.json" # CRA Ignore file to use. If not provided, it checks the repo root directory for `cra-tf-validate-ignore-rules.json`
     PROFILE_ID: "fe96bd4d-9b37-40f2-b39f-a62760e326a3"         # SCC profile ID (currently set to 'IBM Cloud Framework for Financial Services' '1.7.0' profile).
+    CRA_ENVIRONMENT_VARIABLES:
+      TF_VAR_prefix: "test"
```
Comment (jor2) [inline cra-config.yaml:8]: maybe something more accurate like `access-management` or something?

**Review by jor2 — APPROVED** [review]
looks good

File: ibm_catalog.json [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+                "description": "Configures IAM access groups to provide a default set of access policies and assign users to the groups."
+              }
+            ]
+          },
+          "configuration": [
+            {
+              "key": "ibmcloud_api_key"
+            },
+            {
+              "key": "provider_visibility"
```
Comment (jor2) [inline ibm_catalog.json:71]: I've seen some other DA's use this format, maybe we should stay consistent?
````
            {
              "key": "provider_visibility",
              "options": [
                {
                  "displayname": "private",
                  "value": "private"
                },
                {
                  "displayname": "public",
                  "value": "public"
                },
                {
                  "displayname": "public-and-private",
                  "value": "public-and-private"
                }
              ]
            },
````

**Review by jor2 — APPROVED** [review]

**Review by ocofaigh — CHANGES_REQUESTED** [review]
Left some comments. 
Also I need to create a new tile for this, and then you will need to add the `.catalog-onboard-pipeline.yaml` file and update the .releaserc file to onboard to the catalog pipeline. I will do that after your next commit (as I need the offering name updated in the catalog manifest first)

File: ibm_catalog.json [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+      "support_details": "This product is in the community registry, as such support is handled through the originated repo. If you experience issues please open an issue in that repository [https://github.com/terraform-ibm-modules/terraform-ibm-iam-access-group/issues](https://github.com/terraform-ibm-modules/terraform-ibm-iam-access-group/issues). Please note this product is not supported via the IBM Cloud Support Center.",
+      "flavors": [
+        {
+          "label": "Standard",
+          "name": "standard",
+          "install_type": "fullstack",
+          "working_directory": "solutions/access-management",
+          "compliance": {
+            "authority": "scc-v3",
+            "controls": [],
```
Comment (ocofaigh) [inline ibm_catalog.json:43]: remove this line

**Review by ocofaigh — APPROVED** [review]
Lets go with this for now - Suggest to show at next deep dive for feedback

**Pull request merged**